### PR TITLE
Advanced Mapping: add required checkbox to field mapping modal

### PR DESCRIPTION
**Pull request by psadie-sfdo**
# Critical Changes

# Changes
- Adds the "Required" checkbox to the Field Mapping modal
# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by psadie-sfdo** [comment]
**lurch: attach W-036226

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001CB6IQAW" target="_blank">W-036226</a>

**Comment by ptewson-sfdo** [comment]
** lurch: attach W-036226

File: src/lwc/bdiFieldMappingModal/bdiFieldMappingModal.js [hunk 1]
```diff
@@ -390,6 +398,37 @@ export default class bdiFieldMappingModal extends LightningElement {
...
 
+    /*******************************************************************************
+    * @description Converts the "Required__c" field in Data_Import_Field_Mapping__mdt
+    * to a boolean usable in the UI (lightning-input checkbox) or from the UI to a
+    * valid picklist value for use in the metadata record.
+    *
+    * @return {string OR boolean}: Value for the checkbox in the UI or picklist value
+    * for the metadata record
+    */
+    handleIsRequiredField(isChecked) {
```
[the remark this answers is not in the document]
Comment (psadie-sfdo) [reply src/lwc/bdiFieldMappingModal/bdiFieldMappingModal.js]: Yeah, totally makes sense! I was definitely most ify about this part. There's exactly two places this _handleIsRequiredField_ gets called, one for each context you mentioned. I'll split these out. Thanks!!

**Pull request merged**